pkg/ghtransport/ghtransport_test.go

```diff
@@ -0,0 +1,172 @@
+package ghtransport
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"testing"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	"github.com/octo-sts/app/pkg/envconfig"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func TestGCPKMS(t *testing.T) {
+	ctx := context.Background()
+
+	credsFile := createGCPKMSCredsFile(t)
+
+	defer os.Remove(credsFile)
+
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", credsFile)
+	defer os.Unsetenv("GOOGLE_APPLICATION_CREDENTIALS")
+
+	testConfig := envconfig.EnvConfig{
+		Port:            8080,
+		Domain:          "example.com",
+		AppID:           123456,
+		EventingIngress: "https://event.ingress.uri",
+		KMSKey:          "test-kms-key",
+		Metrics:         true,
+	}
+
+	transport, err := New(ctx, testConfig, generateKMSClient(ctx, t))
+
+	assert.NoError(t, err)
+
+	assert.NotNil(t, transport)
+}
+
+func TestCertEnvVar(t *testing.T) {
+	ctx := context.Background()
+
+	testConfig := envconfig.EnvConfig{
+		Port:                       8080,
+		Domain:                     "example.com",
+		AppID:                      123456,
+		EventingIngress:            "https://event.ingress.uri",
+		AppSecretCertificateEnvVar: generateTestCertificateString(),
+		Metrics:                    true,
+	}
+
+	transport, err := New(ctx, testConfig, generateKMSClient(ctx, t))
+
+	assert.NoError(t, err)
+
+	assert.NotNil(t, transport)
+}
+
+func TestCertFile(t *testing.T) {
+	ctx := context.Background()
+
+	testConfig := envconfig.EnvConfig{
+		Port:                     8080,
+		Domain:                   "example.com",
+		AppID:                    123456,
+		EventingIngress:          "https://event.ingress.uri",
+		AppSecretCertificateFile: generateTestCertificateFile(),
+		Metrics:                  true,
+	}
+
+	transport, err := New(ctx, testConfig, generateKMSClient(ctx, t))
+
+	assert.NoError(t, err)
+
+	assert.NotNil(t, transport)
+}
+
+func generateKMSClient(ctx context.Context, t *testing.T) *kms.KeyManagementClient {
+	l, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	fakeServerAddr := l.Addr().String()
+
+	client, err := kms.NewKeyManagementClient(ctx,
+		option.WithEndpoint(fakeServerAddr),
+		option.WithoutAuthentication(),
+		option.WithGRPCDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return client
+}
+
+func createGCPKMSCredsFile(t *testing.T) string {
+
+	tmpFile, err := os.CreateTemp("", "creds-")
```

Comment:
```suggestion
	tmpFile, err := os.CreateTemp(t.TempDir(), "creds-")
```

I know you have an os.Remove above, but this is generally a safer way to ensure the temp file is always cleaned up after the test runs.